File: apps/api/app/schemas/analytics.py
from typing import Optional
from pydantic import BaseModel, ConfigDict


class MetricOverview(BaseModel):
    total_items: int
    total_outfits: int
    utilization_rate: float
    capsule_score: int
    capsule_level: str


class DistributionItem(BaseModel):
    key: str
    label: str
    count: int
    percentage: float
    color: Optional[str] = None


class ColorDistributionItem(BaseModel):
    color_name: str
    hex_code: str
    count: int
    percentage: float
    is_neutral: bool


class VersatileItem(BaseModel):
    id: str
    category: str
    sub_category: str
    primary_color: str
    style: str
    image_url: str
    match_count: int
    versatility_score: float
    highlight_reason: str


class IdleItem(BaseModel):
    id: str
    category: str
    sub_category: str
    primary_color: str
    style: str
    image_url: str
    temp_range: str
    idle_days: int
    revive_suggestion: str


class WardrobeDiagnosis(BaseModel):
    summary: str
    strengths: list[str]
    weaknesses: list[str]
    purchase_suggestions: list[str]


class AnalyticsOverviewResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    metrics: MetricOverview
    categories: list[DistributionItem]
    seasons: list[DistributionItem]
    styles: list[DistributionItem]
    colors: list[ColorDistributionItem]
    neutral_ratio: float
    versatile_items: list[VersatileItem]
    idle_items: list[IdleItem]
    diagnosis: WardrobeDiagnosis

File: apps/web/src/types/analytics.ts
export interface MetricOverview {
  total_items: number;
  total_outfits: number;
  utilization_rate: number;
  capsule_score: number;
  capsule_level: string;
}

export interface DistributionItem {
  key: string;
  label: string;
  count: number;
  percentage: number;
  color?: string;
}

export interface ColorDistributionItem {
  color_name: string;
  hex_code: string;
  count: number;
  percentage: number;
  is_neutral: boolean;
}

export interface VersatileItem {
  id: string;
  category: string;
  sub_category: string;
  primary_color: string;
  style: string;
  image_url: string;
  match_count: number;
  versatility_score: number;
  highlight_reason: string;
}

export interface IdleItem {
  id: string;
  category: string;
  sub_category: string;
  primary_color: string;
  style: string;
  image_url: string;
  temp_range: string;
  idle_days: number;
  revive_suggestion: string;
}

export interface WardrobeDiagnosis {
  summary: string;
  strengths: string[];
  weaknesses: string[];
  purchase_suggestions: string[];
}

export interface AnalyticsOverviewResponse {
  metrics: MetricOverview;
  categories: DistributionItem[];
  seasons: DistributionItem[];
  styles: DistributionItem[];
  colors: ColorDistributionItem[];
  neutral_ratio: number;
  versatile_items: VersatileItem[];
  idle_items: IdleItem[];
  diagnosis: WardrobeDiagnosis;
}
